index.js

```diff
@@ -1,20 +1,236 @@
+    }
+  }
+}
+
+
+let playerInventory = new player (
+  `street`,
+  `empty pocket`
+)
+
+// setting a current room 
+let street = new Room(
+`street`, 
+`You are standing on Main Street between Church and South Winooski.
+There is a door here.`, 
+`door and doormat`,
+`unlocked`
+);
+
+let currentRoom = "street"
+
+let mudRoom = new Room(
+`mudRoom`, 
+`you are standing in a mudroom there are shose and clothes everywhere somone left is a mess... infront of you there is another door! On the left there is another door!`,
+`shoes, clothes and a door`,
+`locked`
+)
+
+let livingRoom = new Room(
+  `living room`,
+  `you have entered the living room it is odly empty, but there are 3 doors on the left, and to the right! what door will you enter?`,
+  `2 doors`,
+  `unlocked`
+)
+
+let garage = new Room(
+  `garage`,
+  `you have entered the garage it is really dark but in the corner there is a bat!`,
+  `bat`,
+  `unlocked`
+)
+
+let basement = new Room(
+  `basement`,
+  `You have entered the basement not much down here other than a old wooden stool`,
+  `stool`,
+  `unlocked`
+)
+
+let bathroom = new Room(
+  `bathroom`,
+  `You have entered the the bathroom, its a normal bathroom but there is a high window!`,
+  `high window`,
+  `unlocked`
+)
+
+let doormat = new Item (
+`door mat`,
+`old most door mat`,
+`You looked under the door mat, there seems to be a hosue key!`,
+false
+)
+
+let houseKey = new Item (
+  `house key`,
+  `shinny house key`,
+  `You have taken the hosue key and put in in you pocket!`,
+  true
+)
+
+let woodenBat = new Item (
+  `bat`,
+  `wooden baseball bat`,
+  `You have picked up the baseball bat!`,
+  true
+)
+
+let stool = new Item(
+  `stool`,
+  `old metal stool`,
+  `You have taken the stool, maybe you can use it to reach somthing!`,
+  true
+)
+
+let lookUpTable = {
+  street: street,
+  mudRoom: mudRoom,
+  livingRoom: livingRoom,
+  garage: garage,
+  basement: basement,
+  bathroom: bathroom,
+  doormat: doormat,
+  houseKey: houseKey,
+  "hosue key": houseKey,
+  woodenBat: woodenBat,
+"wooden bat": woodenBat,
+  stool: stool,
+  playerInventory: playerInventory,
+  currentRoom: currentRoom,
+}
+
+
+// rooms and rooms that you can go into 
+let validRoomChange = {
+  street: ["mudRoom"],
+  mudRoom: ["street", "garage", "livingRoom"],
+  garage: ["mudRoom"],
+  livingRoom:  ["mudRoom","bathRoom", "basement"],
+  basement: ["livingRoom"],
+  bathroom:["livingRoom"]
+};
+ 
 start();
 
 async function start() {
-  const welcomeMessage = `182 Main St.
-You are standing on Main Street between Church and South Winooski.
-There is a door here. A keypad sits on the handle.
-On the door is a handwritten sign.`;
-  let answer = await ask(welcomeMessage);
-  console.log('Now write your code to make this work!');
-  process.exit();
+  let userAction = await ask(`What would you like to do? `);
+
+  let inputArray = incomingInput(userAction)
+
+  let action = inputArray[0]
+
+  let target = inputArray.slice(1).join(" ")
```

Comment:
Your action logic is a little jumbled and I think causing you some trouble. I would suggest setting it up with a new `if...` handling each action, with its `else ifs` handling edge cases and errors. An example would be

```js
if (action === 'use' && lookUpTable[target] instanceof Item) {
//code for successful use
} else if (action === 'use' && !lookUpTable[target] instanceof Item) {
//code for unsuccessful use
}

if (action === 'take' && lookUpTable[target] instanceof Item) {
//code for successful take
} else if (action === 'take' && !lookUpTable[target] instanceof Item) {
//code for unsuccesful take
}
```